src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -78,18 +77,26 @@ protected function action(): Response
         return $this->respondWithData($event->data->object);
     }
 
-    public function handlePaymentIntentSucceeded(Event $event, $donation): Response
+    public function handleChargeSucceeded(Event $event): Response
     {
+        /** @var Donation $donation */
+        $donation = $this->donationRepository->findOneBy(['transactionId' => $event->data->object->payment_intent]);
+
+        if (!$donation) {
+            throw new DomainRecordNotFoundException('Donation not found');
```

Comment:
This should probably info log and return a 204 as well so as not to break multi-sandbox use